src/main/resources/com/openshift/jenkins/plugins/OpenShiftDSL.groovy

```diff
@@ -956,10 +977,20 @@ class OpenShiftDSL implements Serializable {
             if ( objectList != null ) {
                 return objectList;
             }
-
+            
             // Otherwise, we need to ask the API server what presently matches
-            OcAction.OcActionResult r = (OcAction.OcActionResult)script._OcAction( buildCommonArgs("get", selectionArgs(), null, "-o=name") );
-            r.failIf( "Unable to retrieve object names: " + this.toString() );
+            OcAction.OcActionResult r = null;
+            ArrayList<String> g = new ArrayList<String>();
+            g.add("get")
+            OcAction.OcActionResult versionCheck = (OcAction.OcActionResult)script._OcAction( buildCommonArgs("help", g, null, null) );
+            LOGGER.log(Level.FINE, "oc help get output: " + versionCheck.out)
+            if (versionCheck.out.contains("--ignore-not-found")) {
+                r = (OcAction.OcActionResult)script._OcAction( buildCommonArgs("get", selectionArgs(), null, "-o=name", "--ignore-not-found") );
+                r.failIf( "Unable to retrieve object names: " + this.toString() );
+            } else {
+                r = (OcAction.OcActionResult)script._OcAction( buildCommonArgs("get", selectionArgs(), null, "-o=name") );
+                r.failIf( "Unable to retrieve object names: " + this.toString() );
```

Comment:
though `g` in the current form contains 1 entry .... we don't want to do that 

i'll return an empty array list